src/routes/settings/Connections.tsx

```diff
@@ -50,6 +50,24 @@ function mapIntervalToBudgetPeriod(
     }
 }
 
+function mapBudgetRenewalToInterval(
+    budgetRenewal?: string
+): undefined | "Day" | "Week" | "Month" | "Year" {
+    if (!budgetRenewal) return undefined;
+    switch (budgetRenewal) {
+        case "day":
+            return "Day";
+        case "week":
+            return "Week";
+        case "month":
+            return "Month";
+        case "year":
+            return "Year";
+        default:
+            return "Day";
```

Comment:
is having a default safe here? think it might be better to just return undefined